jbpm-designer-backend/src/main/java/org/jbpm/designer/web/server/JbpmServiceRepositoryServlet.java

```diff
@@ -227,22 +248,56 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
 		} 
 	}
 
-	private byte[] getImageBytes(InputStream is) throws Exception {
+	private String getFileIconEncoded(String fileIconPath) {
 		try {
-			return IOUtils.toByteArray(is);
+			return javax.xml.bind.DatatypeConverter.printBase64Binary(
+					getBytesFromFile(new File(fileIconPath.substring(5)))
+			);
+		} catch(Exception e) {
+			try {
+				// return default service icon
+				String defaultServiceNodeIcon = getServletContext().getRealPath(ConfigurationProvider.getInstance().getDesignerContext() + "/defaults/defaultservicenodeicon.png");
+				return javax.xml.bind.DatatypeConverter.printBase64Binary(getBytesFromFile(new File(defaultServiceNodeIcon)));
+			} catch(Exception ee) {
+				_logger.error("Unable to load workitem icon: " + ee.getMessage());
+				return "";
+			}
 		}
-		catch (IOException e) {
-			throw new Exception("Error creating image byte array.");
+	}
+
+	public boolean canUpdateConfigForWorkitem(WorkDefinitionImpl workitem) {
+		return workitem != null &&
+				workitem.getName() != null &&
+				workitem.getName().trim().length() > 0 &&
+				workitem.getDefaultHandler() != null &&
+				workitem.getDefaultHandler().trim().length() > 0;
+	}
+
+	protected static byte[] getBytesFromFile(File file) throws IOException {
```

Comment:
There should be no reason for standard functionality like this to be in the designer esp. as we can use existing libraries already on the classpath.